index.js

```diff
@@ -1,15 +1,24 @@
 'use strict';
 
-var Requested = require('./requested')
-  , https = require('https')
-  , http = require('http');
+module.exports = (function () {
+  if ('XMLHttpRequest' in global) {
+    return require('./browser');
+  }
 
-module.exports = Requested.extend({
-  initialize: function initialize() {
+  var XMLHttpRequest = global.XMLHttpRequest = require('node-http-xhr');
```

Comment:
The global variables could be avoided using dependency injection of some sort, or by duplicating the browser code's feature tests.